thriftpy2/thrift.py

```diff
@@ -68,9 +68,26 @@ def __init__(self):
             pass
         return __init__
 
-    varnames, defaults = zip(*spec)
+    if hasattr(cls, 'thrift_spec'):
+        args = []
+        kwargs = []
+        varnames = []
+        defaults = []
+        for spec_element, t_spec in zip(spec, cls.thrift_spec.values()):
+            varnames.append(spec_element[0])
+            if t_spec[-1]:
+                args.append(spec_element[0])
+            else:
+                kwargs.append(spec_element)
+                defaults.append(spec_element[1])
```

Comment:
This is breaking lot's of valid tests

Reply:
I checked that it is all crashed in Client._send function, I think it is easy to be fixed.

Reply:
@ethe Can you tell me how can I achieve that?

I tried changing args_to_kwargs to not make args to kwargs, passed them to _send, but later failed at `TProcessor:process_in:args = getattr(self._service, api + "_args")()` because of required args and we removed them from defaults.

Reply:
Or we can add a fixture to be added to generated __init__ for raising ValueError if required args don't have value.

Reply:
Compare with raising ValueError in runtime, I think it is better to use args because of a friendly function signature. I will fetch your branch and maybe research for a while. Thank you!

Reply:
@ethe Sure. I also think the same that using `args` is the way to go, I faced blocker here: `Processor` dependency on generated service api. Would love to learn from your implementation.